src/main/native/blake3_jni.cc

```diff
@@ -0,0 +1,74 @@
+// Copyright 2023 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <jni.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "c/blake3.h"
+
+namespace blaze_jni {
+
+jbyte *get_byte_array(JNIEnv *env, jbyteArray java_array) {
+  return env->GetByteArrayElements(java_array, nullptr);
+}
+
+void release_byte_array(JNIEnv *env, jbyteArray array, jbyte *addr) {
+  return env->ReleaseByteArrayElements(array, addr, 0);
+}
+
+extern "C" JNIEXPORT long JNICALL
```

Comment:
This returns `long` while the corresponding Java method returns `int`. It should also the `JNIEnv *` and `jobject` parameters.

Reply:
Done.